x-pack/plugin/vector-tile/src/test/java/org/elasticsearch/xpack/vectortile/feature/FeatureFactoryTests.java

```diff
@@ -242,14 +256,14 @@ public void testManyIntersectingGeometries() {
                 expandByHalf(GeoTileUtils.toBoundingBox(2 * x + 1, 2 * y, z + 1))
             )
         );
-        assertThat(builder.getFeatures(intersectCollection), iterableWithSize(2));
+        assertThat(builder.getFeatures(intersectCollection), iterableWithSize(1));
         // contain geometries
         assertThat(builder.getFeatures(GeoTileUtils.toBoundingBox(x / 4, y / 4, z - 2)), iterableWithSize(1));
         assertThat(builder.getFeatures(GeoTileUtils.toBoundingBox(x / 4, y / 4, z - 2)), iterableWithSize(1));
         final GeometryCollection<Geometry> containCollection = new GeometryCollection<>(
             List.of(GeoTileUtils.toBoundingBox(x / 4, y / 4, z - 2), GeoTileUtils.toBoundingBox(x / 4, y / 4, z - 2))
         );
-        assertThat(builder.getFeatures(containCollection), iterableWithSize(2));
+        assertThat(builder.getFeatures(containCollection), iterableWithSize(1));
```

Comment:
Why only `1`? I would expect this to remain `2`. Or is it that we no longer flatten to a list, but keep the geometry collection?

Reply:
see comment above